### PR TITLE
Add common observability code

### PR DESCRIPTION
**Pull request by yegle**

### CODE REVIEW

**Comment by mikehelmick** [comment]
see build errors

**Review by sethvargo — COMMENTED** [review]
@yegle I think this is already happening? We just refactored to use the o11y middleware and I _think_ it's doing the right thing.

**Comment by yegle** [comment]
> @yegle I think this is already happening? We just refactored to use the o11y middleware and I _think_ it's doing the right thing.

The `WithBuildInfo` is hardcoded to use key server's build info. There's a slight change here to make it more general.

Let me split this to two PRs. It gets a bit confusing.

**Review by sethvargo — COMMENTED** [review]
Are the go mod changes intentional?

File: pkg/observability/common.go [hunk 1]
```diff
@@ -14,8 +14,95 @@
...
+)
+
+var (
+	// ResultOK add a tag indicating the API call is a success.
+	ResultOK = tag.Upsert(ResultTagKey, "OK")
+	// ResultNotOK add a tag indicating the API call is a failure.
+	ResultNotOK = ResultError("NOT_OK")
+)
+
+var (
```
Comment (sethvargo) [inline pkg/observability/common.go:76]: I think a lot of this is already defined in https://github.com/google/exposure-notifications-server/blob/main/pkg/observability/observability.go. We may want to combine/reconcile the two files.

Comment (sethvargo) [reply pkg/observability/common.go:76]: https://github.com/google/exposure-notifications-server/blob/main/pkg/observability/observability.go#L37

Comment (yegle) [reply pkg/observability/common.go]: Yes, those are removed.

**Review by sethvargo — COMMENTED** [review]
Sorry, I should have clarified. I think it's weird that we have constants across observability.go and common.go. I think we should merge the contents of both files into 1, pref observability.go since it follows the naming pattern of other packages.

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**